### PR TITLE
Header file for CLI test class

### PR DESCRIPTION
**Pull request by joergi-w**
In the first commit I move the CLI test class to its own header file. This enables the class to be included and used by multiple test files. 

I found that the file names for the tests were too generic. In the 2nd commit I rename `cli_test` to `fasta_to_fastq_options_test` because it tests the options of the program. The `fastq_conversion_test` becomes `convert_fastq_test` because the function to be tested in this API test has this name.

Additional remark:
I read in the [Gtest documentation](https://github.com/google/googletest/blob/master/googletest/docs/faq.md#why-should-test-suite-names-and-test-names-not-contain-underscore) that test names should not contain underscores. In SeqAn3 we use snake_case for test names, which contain a lot of underscores. I discussed with marehr that we keep underscores for now (e.g. `cli_test` or `out_not_empty`). However, I would be happy to have a reviewer's opinion about this as well.

### CODE REVIEW

File: test/cli/README.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
-# App Test
+# CLI Test
 
-Here are test files for blackbox tests, i.e. the app is executed with defined input files and parameters. The test then validates whether the output is correct.
+Here are test files for command line interface tests, i.e. the app is executed with defined input files and parameters.
+The test then validates whether the output is correct.
 
+Each test should be inherited from the `cli_test` class: It provides the functionality of executing the app,
```
Comment (smehringer) [inline test/cli/README.md:6]:
```suggestion
Each test fixture should be inherited from the `cli_test` class: It provides the functionality of executing the app, 
```
?

File: test/cli/cli_test.hpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        // Run the command and return results.
+        result.exit_code = std::system(command.str().c_str());
+        result.out = testing::internal::GetCapturedStdout();
+        result.err = testing::internal::GetCapturedStderr();
+        return result;
+    }
+
+    // Generate the full path of a test input file that is provided in the data directory.
+    static
+    std::filesystem::path data(std::string const & filename)
```
Comment (smehringer) [inline test/cli/cli_test.hpp:54]: on same line as static? :nail_care:

File: test/cli/README.md (hunk 1 above)
Comment (joergi-w) [reply test/cli/README.md:6]: Yes 👍

**Comment by joergi-w** [comment]
> What is the rational of Gtest? I think it looks weird if we start using camelCase in test names.. but if it avoids error I'm not sure

They forbid underscore as first and last character and do not recommend it in the middle. Their macro chains the name parts together to `TestSuiteName_TestName_Test`, which can become ambiguous with underscores in the names. The last sentence says it well:

> If you violate the rule, there may not be immediate consequences, but your test may (just may) break with a new compiler (or a new version of the compiler you are using) or with a new version of googletest. Therefore it's best to follow the rule.

I guess we can leave it for now and think about it in a separate issue.

**Comment by joergi-w** [comment]
Thanks! I have opened the issue here: https://github.com/seqan/product_backlog/issues/76

**Pull request merged**